x/wasm/internal/keeper/options.go

```diff
@@ -1,28 +1,73 @@
 package keeper
 
-import "github.com/CosmWasm/wasmd/x/wasm/internal/types"
+import (
+	"fmt"
+	"github.com/CosmWasm/wasmd/x/wasm/internal/types"
+)
 
 type optsFn func(*Keeper)
 
 func (f optsFn) apply(keeper *Keeper) {
 	f(keeper)
 }
 
-// WithMessageHandler is an optional constructor parameter to replace the default wasm vm engine with the
+// WithMessageHandler is an optional constructor parameter to replace the default wasmVM engine with the
 // given one.
 func WithWasmEngine(x types.WasmerEngine) Option {
 	return optsFn(func(k *Keeper) {
-		k.wasmer = x
+		k.wasmVM = x
 	})
 }
 
-// WithMessageHandler is an optional constructor parameter to set a custom message handler.
+// WithMessageHandler is an optional constructor parameter to set a custom handler for wasmVM messages.
```

Comment:
Not a blocker, but an idea for the future.

Now that we chain the handlers, most customizations (outside of tests) would likely be prepending or appending something to the default chain. This should be easy to do. 

Right now, making a new chain like the default is exposed as public constructors

```go
return NewMessageHandlerChain(
		NewSDKMessageHandler(router, encoders),
		NewIBCRawPacketHandler(channelKeeper, capabilityKeeper),
	)
```

But it does require the router, and various keepers to construct it. Those are not exposed on the keeper, we could only extend the default if we do that in `app.go` where we have the keepers, or some other code that wraps the constuctor of the keeper. Maybe that is the only place it would make sense to customize the handler, but just thinking....

If this makes sense to you, please open a future issue to allow appending/prepending to the default chain. If you think this is do-able with the current API, then I am fine.

Use case: if there is something I cannot express as an Encoder and I want to add support for that Custom message, I want to be able to extend this.